### ISSUE TITLE
obsoletion of grouping terms for "cofactor" & "coenzyme" roles

### ISSUE THREAD

**Issue by krchristie** [geneontology/go-ontology#19243]
Based on discussion at Ontology Editors calls on 3/30/30 & 4/6/20, with follow-on discussion at the Annotation Call on 4/7/20, we are going to obsolete all of the terms that refer to the ChEBI roles of "cofactor" or "coenzyme" in their equivalence axioms as well as other grouping terms that reference those terms, basically all of the grouping terms for "cofactor" and "coenzyme".  The reasons are
- Reasoning over ChEBI roles is producing many inferences that are not appropriate in GO.
- It is not very meaningful to group processes that are not metabolically related by the fact that they produce a compound that may be used as a cofactor or to group various kinds of binding to chemically unrelated compounds by the fact that they may be used as cofactors.

The existing child terms will remain under their existing chemical classifications.

These are the `cofactor` terms slated for obsoletion, along with the counts of direct annotations that need to be reviewed, as well as whether the term has been used for IBA annotations of Panther families:

GO_term | # annots w/o IBA IEA | # EXP | # IC/ TAS/ NAS | # SEQ | Panther?
-- | -- | -- | -- | -- | --
cofactor metabolic process | 16 | 1 | 9 | 6 | Y
cofactor biosynthetic process | 4 | 2 | - | 2 | Y
cofactor catabolic process | 12 | 12 | - | - | -
protein-cofactor   linkage | 3 | 3 | - | - | -
cofactor binding | 51 | 32 | - | 19 | -
cofactor transport | 7 | 1 | - | 6 | -
cofactor transmembrane transporter activity | 0 | 0 | - | - | Y
regulation of cofactor metabolic process | - | - | - | - | -
negative regulation of cofactor metabolic process | - | - | - | - | -
positive regulation of cofactor metabolic process | - | - | - | - | -
Totals  | 93 | 51 | 9 | 33 |  

These are the `coenzyme` terms slated for obsoletion, with the same format as above:

GO_term | w/o IBA IEA | EXP | IC/ TAS/ NAS | RCA | SEQ | Panther?
-- | -- | -- | -- | -- | -- | --
coenzyme   binding | 36 | 14 | - | - | 22 |  
coenzyme   metabolic process | - | - | - | - | 3 |  
coenzyme   biosynthetic process | 30 | 11 | 6 | 6 | 7 | Y
coenzyme   catabolic process | - | - | - | - | - |  
coenzyme   transport | 0 | - | - | - | - |  
coenzyme   transmembrane transporter activity | - | - | - | - | - |  
regulation of   coenzyme metabolic process | - | - | - | - | - |  
negative   regulation of coenzyme metabolic process | - | - | - | - | - |  
positive   regulation of coenzyme metabolic process | - | - | - | - | - |  
Totals  | 66 | 25 | 6 | 6 | 32 |  

This plan to obsolete these terms from the ontology replaces the previous idea to remove the logical definitions of these terms: **removal of cofactor & coenzyme roles from equivalence axioms** https://github.com/geneontology/go-ontology/issues/17911

**Comment by krchristie** [comment]
**Email for obsoletion of cofactor grouping terms** - _sent 4/13/2020_

Dear all, 
 
The proposal has been made to obsolete the cofactor grouping terms:
- GO:0051186 - cofactor metabolic process
- GO:0051188 - cofactor biosynthetic process
- GO:0051187 - cofactor catabolic process
- GO:0018065 - protein-cofactor linkage
- GO:0048037 - cofactor binding
- GO:0051181 - cofactor transport
- GO:0051184 - cofactor transmembrane transporter activity
- GO:0051193 - regulation of cofactor metabolic process
- GO:0051195 - negative regulation of cofactor metabolic process
- GO:0051194 - positive regulation of cofactor metabolic process
 
The reasons for these obsoletions are:
- Reasoning over the ChEBI role of “cofactor” is producing many inferences that are not appropriate in GO.
- It is not very meaningful to group processes that are not metabolically related by the fact that they produce a compound that may be used as a cofactor or to group various kinds of binding to chemically unrelated compounds by the fact that they may be used as a cofactor.
 
There are 60 annotations to these terms that need to be reviewed and 33 sequence annotations that may need to be reviewed: 

Contributor | EXP + AS | SEQ
-- | -- | --
AgBase | 0 | 2
BHF-UCL | 1 | 0
CGD | 1 | 0
EcoCyc | 3 | 0
EcoliWiki | 1 | 0
GeneDB | 4 | 0
MGI | 0 | 2
PomBase | 0 | 1
PseudoCAP | 0 | 2
Reactome | 9 | 0
RGD | 7 | 2
SGD | 3 | 0
UniProt | 31 | 23
ZFIN | 0 | 1
Grand Total | 60 | 33

There are InterPro2GO & UniRule mappings to these terms: 

GO_term | external mappings
-- | --
cofactor metabolic process | InterPro:IPR010722, InterPro:IPR017678, InterPro:IPR017680
cofactor biosynthetic process | InterPro:IPR003203,InterPro:IPR008225, InterPro:IPR010115, InterPro:IPR011774, InterPro:IPR013477, InterPro:IPR013480, InterPro:IPR023659, InterPro:IPR023661, UniRule:UR000028575, UniRule:UR000195377, UniRule:UR000195519, UniRule:UR000375818
protein-cofactor linkage | InterPro:IPR016614
cofactor binding | InterPro:IPR003781, InterPro:IPR006140, InterPro:IPR011159, InterPro:IPR012349, InterPro:IPR015761, UniRule:UR000171791, UniRule:UR000638782, UniRule:UR000925792

These terms are included in these subset(s).

GO_term | subset
-- | --
cofactor metabolic process | chembl, drosophila, generic, pir, pombe, yeast
cofactor binding | pir
cofactor transport | pir

Please consider if these annotations can be transferred to a term for some specific type of cofactor. Let me know if you think we may need any new terms.
 
Any comments can be  added to the issue.
 
We are opening a comment period for these proposed obsoletions. We’d like to proceed and obsolete this term on April 27th. **Unless objections are received by April 27th, we will assume that you agree to this change.**

**Comment by krchristie** [comment]
**Email for obsoletion of coenzyme grouping terms** - _sent 4/13/2020_

Dear all, 
 
The proposal has been made to obsolete the coenzyme grouping terms:
-	GO:0050662 - coenzyme binding
-	GO:0006732 - coenzyme metabolic process
-	GO:0009108 - coenzyme biosynthetic process
-	GO:0009109 - coenzyme catabolic process
-	GO:0051182 - coenzyme transport
-	GO:0051185 - coenzyme transmembrane transporter activity
-	GO:0051196 - regulation of coenzyme metabolic process
-	GO:0051198 - negative regulation of coenzyme metabolic process
-	GO:0051197 - positive regulation of coenzyme metabolic process
 
The reasons for these obsoletions are:
-	Reasoning over ChEBI role of “coenzyme” is producing many inferences that are not appropriate in GO.
-	It is not very meaningful to group processes that are not metabolically related by the fact that they produce a compound that may be used as a coenzyme or to group various kinds of binding to chemically unrelated compounds by the fact that they may be used as a coenzyme.

There are 37 annotations to these terms that need to be reviewed and 32 sequence annotations that may need to be reviewed: 

Contributor | EXP TAS RCA | SEQ
-- | -- | --
AspGD | 7 | 0
CAFA | 2 | 0
EcoCyc | 2 | 0
GeneDB | 0 | 1
JCVI | 0 | 4
MGI | 1 | 8
MTBBASE | 4 | 0
PseudoCAP | 4 | 0
Reactome | 6 | 0
RGD | 4 | 6
TIGR | 0 | 1
UniProt | 7 | 12
Grand Total | 37 | 32

There are InterPro2GO & UniRule mappings to these terms: 

GO Term | External Mapping
-- | --
coenzyme binding | InterPro:IPR001509, InterPro:IPR002202, InterPro:IPR003421, InterPro:IPR004553, InterPro:IPR009023, InterPro:IPR009029, InterPro:IPR013118, UniRule:UR000100714
coenzyme biosynthetic process | InterPro:IPR002835, InterPro:IPR005980, InterPro:IPR019939, UniRule:UR000039422, UniRule:UR000039429, UniRule:UR000079254, UniRule:UR000079257, UniRule:UR000092052, UniRule:UR000101490

The term “**coenzyme binding**” is included in the subset: **metagenomics**.
 
Please consider if these annotations can be transferred to a term for some specific type of cofactor. Let me know if you think we may need any new terms.
 
Any comments can be  added to the issue.
 
We are opening a comment period for these proposed obsoletions. We’d like to proceed and obsolete this term on April 27th. **Unless objections are received by April 27th, we will assume that you agree to this change.**

**Comment by krchristie** [comment]
InterPro curators: @lsw26 @sarach06 @happy-lorna @EBI-Hsinyu - Please take a look at the InterPro mappings to the cofactor and coenzyme grouping terms listed above.

**Comment by ValWood** [comment]
hi @krchristie 
I can't locate 
PomBase | 0 | 1

copuld you provide a breakdown by term?

tks
v

**Comment by EBI-Hsinyu** [comment]
Hi Karen @krchristie , 
Thank you for letting us know. I have removed these GO terms from InterPro.
Regards, 
Hsin-Yu

**Comment by mah11** [comment]
@ValWood - it's SPAC4H3.09, annotated to GO:0048037 by ISM with Pfam:PF00550

**Comment by ValWood** [comment]
fixed thanks

**Comment by Antonialock** [comment]
> fixed thanks

now I went on a goose chase . for this annotation :) marked as removed in spreadsheet

**Comment by RLovering** [comment]
are you going to create a google spreadsheet so that the review of these annotations can be tracked?

**Comment by Antonialock** [comment]
There's another ticket @RLovering 
https://docs.google.com/spreadsheets/d/15aLkDsoScybffpARFmbrZxO0Ki8z7YVU_epgyvBwBjs/edit#gid=0

**Comment by RLovering** [comment]
Thanks  ;) I had done ours already, very difficult to keep track

**Comment by krchristie** [comment]
Completed all obsoletions on 5/5/20

additional notes
- The term `oxidoreduction coenzyme metabolic process` (GO:0006733) was also obsoleted. Since it didn't have any direct annotations in either AmiGO or QuickGO, I went ahead and obsoleted it along with the rest of this set.

- There was an odd situation when trying to obsolete/deprecate `coenzyme metabolic process (`. It was referred to as the replacement term for the obsolete term GO:0006752, which had also already been merged into `coenzyme metabolic process (GO:0006732)`, which meant that GO:0006752 was an alt_id for GO:0006732. The Usages tab indicated that GO:0006732 was used as a replaced_by term for GO:0006752 and when I clicked through to that usage, I would see this:

![replacedByToCoenzymeMP](https://user-images.githubusercontent.com/10533218/81239094-b0221700-8fb8-11ea-9948-9b30c4cf4b3f.png)

If I deleted the replaced_by annotation from GO_0006752, Protege would let me save the file, but then the diff would show the addition of this incomplete stanza for the the obsolete term that had also been merged into `coenzyme metabolic process (GO:0006732)`.

> GO:0006752
> [Term]
> id: GO:0006752
> property_value: IAO:0000231 IAO:0000227
> is_obsolete: true

If I removed this incomplete stanza, then there were no differences in the go-edit.obo file to commit.

Since the GO:0006752 ID was already an alt_id for `coenzyme metabolic process (GO:0006732)`, it seems like it was a mistake to have the replaced_by link and I just obsoleted `coenzyme metabolic process (GO:0006732)`, and this passed the travis checks so I committed this way. I wanted to note this situation in case any errors come up later.
